src/operations/insert.ts

```diff
@@ -135,21 +135,24 @@ export class InsertManyOperation extends AbstractCallbackOperation<InsertManyRes
       options
     );
 
-    bulkWriteOperation.executeCallback(server, session, (err, res) => {
+    let res;
+    try {
+      res = await bulkWriteOperation.execute(server, session);
+    } catch (err) {
       if (err || res == null) {
         if (err && err.message === 'Operation must be an object with an operation key') {
-          err = new MongoInvalidArgumentError(
+          throw new MongoInvalidArgumentError(
             'Collection.insertMany() cannot be called with an array that has null/undefined values'
           );
         }
-        return callback(err);
+        return err;
       }
-      callback(undefined, {
-        acknowledged: writeConcern?.w !== 0 ?? true,
-        insertedCount: res.insertedCount,
-        insertedIds: res.insertedIds
-      });
-    });
+    }
+    return {
+      acknowledged: writeConcern?.w !== 0,
+      insertedCount: res.insertedCount,
+      insertedIds: res.insertedIds
+    };
   }
```

Comment:
we can clean this up
- we only need to check if `res` is null when we're using callbacks, because the signature for callbacks is `(error?, result?) => void`.  
- that means we can pull `res` into the try block specifically and pull the `return` statement into the `try` block too
- we need to `throw err`, not `return err` on line 148

Reply:
if we put the return in the try block, the function throws saying that it's lacking a return statement

Reply:
sorry, yes, we can also remove the `if(err)` too and then we should always be throwing from the `catch` block